python/lsst/pipe/tasks/assembleCoadd.py

```diff
@@ -1746,6 +1766,32 @@ def findArtifacts(self, templateCoadd, tempExpRefList, imageScalerList):
                              'EDGE': edge})
         return altMasks
 
+    def _prefilterArtifacts(self, spanSetList, exp):
+        """!
+        \brief Remove artifact candidates covered by bad mask plane
+
+        Any future editing of the candidate list that does not depend on
+        temporal information should go in this method.
+
+        @param spanSetList: List of SpanSets representing artifact candidates
+        @param exp: Exposure containing mask planes used to prefilter
+
+        return List of SpanSets with artifacts
+        """
+        badPixelMask = afwImage.Mask.getPlaneBitMask(self.config.prefilterDefectsMaskPlanes)
+        goodArr = numpy.where(exp.mask.array & badPixelMask, False, True)
+        returnSpanSetList = []
+        bbox = exp.getBBox()
+        x0, y0 = exp.getXY0()
+        for i, span in enumerate(spanSetList):
+            y, x = span.clippedTo(bbox).indices()
+            yIdxLocal = [y1 - y0 for y1 in y]
+            xIdxLocal = [x1 - x0 for x1 in x]
```

Comment:
Aren't `x` and `y` numpy arrays, so you can just do:

    yIdxLocal = y - y0
    xIdxLocal = x - x0

If they're not numpy arrays, just cast them.

And I think you should spell out `xIndexLocal`.

Reply:
This failed a unit test. Reverting. 

            y, x = span.clippedTo(bbox).indices()
            yIndexLocal = numpy.array(y) - y0
            xIndexLocal = numpy.array(x) - x0
>           goodRatio = numpy.count_nonzero(goodArr[yIndexLocal, xIndexLocal])/span.getArea()
E           TypeError: tuple indices must be integers or slices, not tuple